### PR TITLE
Upgrade to Laravel 6.0.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request upgrades Phoenix to Laravel 6.0, the latest long-term service release.

### How should this be reviewed?

:six: Updated Laravel 5.5 to Laravel 5.6, following the [upgrade guide](https://laravel.com/docs/5.6/upgrade) in cf335e75

:seven: Updated Laravel 5.6 to Laravel 5.7, following the [upgrade guide](https://laravel.com/docs/5.7/upgrade) in b44c2e83

:eight: Updated Laravel 5.7 to Laravel 5.8, following the [upgrade guide](https://laravel.com/docs/5.8/upgrade) in 7537b335. This also required updating to [Gateway 3.0](https://github.com/DoSomething/gateway/releases/tag/v3.0.0).

:zero: Updated Laravel 5.8 to Laravel 6.0, following the [upgrade guide](https://laravel.com/docs/6.0/upgrade) in aca45ca1. This also involved updating the [contentful/contentful](https://github.com/contentful/contentful.php/blob/master/CHANGELOG.md), [contentful/laravel](https://github.com/contentful/contentful-laravel/blob/master/CHANGELOG.md), and [dfurnes/environmentalist](https://github.com/DFurnes/environmentalist/releases/tag/v0.0.6) packages to the latest versions.

:pencil2:  Finally, I've updated files (like config values and comments) to match the [`laravel/laravel`](https://github.com/laravel/laravel/tree/6.x) repository in 462d899. You can see the corresponding upstream changes in this [absolute unit of a diff](https://github.com/laravel/laravel/compare/5.5...6.x). For the most part, these are just stylistic updates.

### Any background context you want to provide?

This is part of our "Q3 Upgrades" epic, and ensures that we keep getting security and bug fixes! It also includes new features, so it's worth skimming the release notes for [Laravel 5.6](https://laravel.com/docs/5.6/releases#laravel-5.6), [Laravel 5.7](https://laravel.com/docs/5.7/releases#laravel-5.7), [Laravel 5.8](https://laravel.com/docs/5.8/releases#laravel-5.8) and [Laravel 6.0](https://laravel.com/docs/6.x/releases#laravel-6) to see what's new! 🆕 

### Relevant tickets

References [Pivotal #172383659](https://www.pivotaltracker.com/story/show/172383659).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: app/Http/Middleware/TrustProxies.php [hunk 1]
```diff
@@ -16,15 +16,9 @@ class TrustProxies extends Middleware
...
+     * @var int
      */
-    protected $headers = [
-        Request::HEADER_FORWARDED => null, // Not set on AWS or Heroku.
-        Request::HEADER_X_FORWARDED_FOR => 'X_FORWARDED_FOR',
-        Request::HEADER_X_FORWARDED_HOST => null, // Not set on AWS or Heroku.
-        Request::HEADER_X_FORWARDED_PORT => 'X_FORWARDED_PORT',
-        Request::HEADER_X_FORWARDED_PROTO => 'X_FORWARDED_PROTO',
-    ];
+    protected $headers = Request::HEADER_X_FORWARDED_AWS_ELB;
```
Comment (DFurnes) [inline app/Http/Middleware/TrustProxies.php:23]: Note: We're not using ELB, but this is still the [suggested setting](https://devcenter.heroku.com/articles/getting-started-with-laravel#trusting-the-load-balancer) in Heroku's setup guide.

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -48,7 +48,7 @@ public function boot(Router $router)
                 'SIXPACK_COOKIE_PREFIX' => config('services.sixpack.prefix'),
                 'SIXPACK_ENABLED' => config('services.sixpack.enabled'),
                 'SIXPACK_TIMEOUT' => config('services.sixpack.timeout'),
-                'CONTENTFUL_USE_PREVIEW_API' => config('contentful')['delivery.preview'],
+                'CONTENTFUL_USE_PREVIEW_API' => config('contentful.delivery.preview'),
```
Comment (DFurnes) [inline app/Providers/AppServiceProvider.php:51]: This syntax always bothered me, so I'm relieved to see they've made this more standard in the latest release!

File: .phpunit.result.cache [hunk 3]
```diff
@@ -0,0 +1 @@
...ntity_is_required_if_field_is_displayed_to_submit_photo_post";d:0.051;s:97:"Tests\Feature\StorePhotoReportbackPostTest::quantity_must_be_a_valid_integer_to_submit_photo_post";d:0.058;s:96:"Tests\Feature\StorePhotoReportbackPostTest::quantity_must_be_a_whole_number_to_submit_photo_post";d:0.052;s:92:"Tests\Feature\StorePhotoReportbackPostTest::quantity_must_be_at_least_1_to_submit_photo_post";d:0.051;s:87:"Tests\Feature\StorePhotoReportbackPostTest::file_media_is_required_to_submit_photo_post";d:0.055;s:103:"Tests\Feature\StorePhotoReportbackPostTest::file_media_must_be_a_valid_file_upload_to_submit_photo_post";d:0.039;s:97:"Tests\Feature\StorePhotoReportbackPostTest::file_media_must_be_a_valid_image_to_submit_photo_post";d:0.035;s:93:"Tests\Feature\StorePhotoReportbackPostTest::why_participated_is_required_to_submit_photo_post";d:0.05;s:81:"Tests\Feature\StorePhotoReportbackPostTest::registered_user_can_submit_photo_post";d:0.036;s:83:"Tests\Feature\StorePhotoReportbackPostTest::anonymous_user_cannot_submit_photo_post";d:0.04;s:66:"Tests\Feature\ViewCampaignsTest::user_can_view_a_list_of_campaigns";d:0.042;s:57:"Tests\Feature\ViewCampaignsTest::user_can_view_a_campaign";d:0.039;}}}
```
Comment (weerd) [inline .phpunit.result.cache:1]: Should this be in the `.gitignore`?

File: app/Repositories/QueriesContentful.php [hunk 4]
```diff
@@ -24,11 +25,11 @@ protected function getEntriesAsJson($type, $options = [])
     {
         $query = (new Query)
                 ->setContentType($type)
-                ->setInclude(array_get($options, 'includeDepth', 0))
+                ->setInclude(arr::get($options, 'includeDepth', 0))
```
Comment (weerd) [inline app/Repositories/QueriesContentful.php:28]: Typo! Capitalize the `A` in `Arr`.

File: app/helpers.php [hunk 5]
```diff
@@ -43,11 +44,10 @@ function get_cache_url($prefix, $slug = null)
...
 function get_contentful_url()
 {
-    $contentful = config('contentful');
-    $spaceId = $contentful['delivery.space'];
-    $envronment = $contentful['delivery.environment'];
+    $spaceId = config('contentful.delivery.space');
+    $environment = config('contentful.delivery.environment');
 
-    $environmentPath = $envronment !== 'master' ? 'environments/'.$envronment.'/' : '';
+    $environmentPath = $environment !== 'master' ? 'environments/'.$environment.'/' : '';
```
Comment (weerd) [inline app/helpers.php:50]: We had it spelled wrong this WHOLE time @_@

File: composer.json [hunk 6]
```diff
@@ -4,68 +4,70 @@
...
-    "symfony/css-selector": "3.1.*",
-    "symfony/dom-crawler": "3.1.*",
-    "laravel/browser-kit-testing": "2.*"
+    "laravel/tinker": "^2.0",
+    "mockery/mockery": "^1.0",
+    "phpunit/phpunit": "^8.0",
+    "symfony/css-selector": "^4.0",
+    "symfony/dom-crawler": "^4.0",
+    "nunomaduro/collision": "^3.0",
+    "laravel/browser-kit-testing": "^5.0"
```
Comment (weerd) [inline composer.json:29]: Do we wanna add `"beyondcode/laravel-dump-server"`? I saw that Taylor removed it in 5.9 due to something not working, but I tested it in the 6.0 Rogue updates and works well. Not sure if something in (unreleased) 5.9 had been conflicting with it but then was fine after further changes for 6.0. 

I find the `dump-server` super helpful, especially when debugging stuff for React front-end.

File: public/svg/403.svg [hunk 7]
```diff
@@ -0,0 +1 @@
..."/><circle cx="623" cy="225" r="3" fill="#FFFFFF"/><circle cx="823" cy="55" r="3" fill="#FFFFFF"/><circle cx="902" cy="144" r="2" fill="#FFFFFF"/><circle cx="552" cy="14" r="2" fill="#FFFFFF"/></g><path fill="#486587" d="M796 535a4 4 0 0 1 4 4v20h-8v-20a4 4 0 0 1 4-4z"/><path fill="#071423" d="M798 535.54a4 4 0 0 0-2 3.46v20h-4v-20a4 4 0 0 1 6-3.46zm48-.54a4 4 0 0 1 4 4v20h-8v-20a4 4 0 0 1 4-4z"/><path fill="#8DBCD2" d="M846 559v-20a4 4 0 0 0-2-3.46 4 4 0 0 1 6 3.46v20h-4z"/><g fill="#FFFFFF" opacity=".07" transform="translate(54 301)"><path d="M554.67 131.48a9.46 9.46 0 0 1 13.33 0 9.46 9.46 0 0 0 13.33 0l13.33-13.24a28.39 28.39 0 0 1 40 0l10 9.93a14.2 14.2 0 0 0 20 0 14.2 14.2 0 0 1 20 0l.6.6a31.8 31.8 0 0 1 9.4 22.56H548v-3.84c0-6.01 2.4-11.78 6.67-16.01zM751 8.25c11.07-11 28.93-11 40 0l10 9.94a14.19 14.19 0 0 0 20 0 14.19 14.19 0 0 1 20 0 16.36 16.36 0 0 0 21.3 1.5l8.7-6.47a33.47 33.47 0 0 1 40 0l4.06 3.03A39.6 39.6 0 0 1 931 48H731c0-12.72 8.93-28.75 20-39.75zM14.1 75.14l.9-.9a21.29 21.29 0 0 1 30 0 21.29 21.29 0 0 0 30 0l10-9.93a35.48 35.48 0 0 1 50 0l15 14.9a14.2 14.2 0 0 0 20 0 14.2 14.2 0 0 1 20 0c6.4 6.35 10 15 10 24.02V109H0c0-12.71 5.07-24.9 14.1-33.86z"/></g></g></svg>
```
Comment (weerd) [inline public/svg/403.svg:1]: The `public/svg` directory and  SVG files were added in like 5.8 but then removed by 6.0.

File: public/svg/404.svg [hunk 8]
```diff
@@ -0,0 +1 @@
...ill="url(#e)" d="M1024 285.32V412H857c99.31-86.6 112.63-140.94 39.97-163L1024 285.32z"/><path fill="url(#f)" d="M0 474V223.93C67.12 190.69 129.55 155 263 155c250 0 331.46 162.6 530 175 107.42 6.71 163-26.77 231-58.92V474H0z"/><path fill="url(#e)" d="M353.02 474H0V223.93C67.12 190.69 129.55 155 263 155c71.14 0 151.5 12.76 151.5 70.5 0 54.5-45.5 79.72-112.5 109-82.26 35.95-54.57 111.68 51.02 139.5z"/><path fill="url(#g)" d="M353.02 474H0v-14.8l302-124.7c-82.26 35.95-54.57 111.68 51.02 139.5z"/></g><g fill="#FFFFFF" opacity=".2" transform="translate(288 523)"><circle cx="250" cy="110" r="110"/><circle cx="420" cy="78" r="60"/><circle cx="70" cy="220" r="70"/></g><g fill="#FFFFFF" fill-rule="nonzero" opacity=".08" transform="translate(135 316)"><path d="M10 80.22a14.2 14.2 0 0 1 20 0 14.2 14.2 0 0 0 20 0l20-19.86a42.58 42.58 0 0 1 60 0l15 14.9a21.3 21.3 0 0 0 30 0 21.3 21.3 0 0 1 30 0l.9.9A47.69 47.69 0 0 1 220 110H0v-5.76c0-9.02 3.6-17.67 10-24.02zm559.1-66.11l5.9-5.86c11.07-11 28.93-11 40 0l10 9.94a14.19 14.19 0 0 0 20 0 14.19 14.19 0 0 1 20 0 16.36 16.36 0 0 0 21.3 1.5l8.7-6.47a33.47 33.47 0 0 1 40 0l4.06 3.03A39.6 39.6 0 0 1 755 48H555a47.77 47.77 0 0 1 14.1-33.89z"/></g></g></svg>
```
Comment (weerd) [inline public/svg/404.svg:1]: The `public/svg` directory and  SVG files were added in like 5.8 but then removed by 6.0.

File: public/svg/500.svg [hunk 9]
```diff
@@ -0,0 +1 @@
... 2011.85 -1427.831)"><path fill="#FFFFFF" d="M13.54 2.58a3.46 3.46 0 0 1-.82 5.5A17.18 17.18 0 0 0 9.3 10.2c-.41.42-.94 1.12-1.58 2.1a2.82 2.82 0 0 1-4.36.45 2.82 2.82 0 0 1 .45-4.36c.99-.64 1.68-1.17 2.1-1.58.46-.46 1.17-1.6 2.13-3.42a3.46 3.46 0 0 1 5.5-.82z" opacity=".6"/><path fill="url(#j)" d="M8.26 4.8a3.46 3.46 0 0 1 3.3 4.47c-.6 1.97-.9 3.27-.9 3.93 0 .59.12 1.45.36 2.6a2.82 2.82 0 0 1-2.76 3.4 2.82 2.82 0 0 1-2.76-3.4c.24-1.15.36-2.01.36-2.6 0-.66-.3-1.96-.9-3.93a3.46 3.46 0 0 1 3.3-4.47z" transform="rotate(45 8.258 12)"/></g><circle cx="756" cy="209" r="110" fill="#FFFFFF" opacity=".2"/><circle cx="859" cy="139" r="40" fill="#FFFFFF" opacity=".2"/><circle cx="551" cy="383" r="70" fill="#FFFFFF" opacity=".2"/><circle cx="666" cy="359" r="30" fill="#FFFFFF" opacity=".2"/><rect width="60" height="6" x="722" y="547" fill="#FFFFFF" opacity=".4" rx="3"/><rect width="60" height="6" x="842" y="565" fill="#FFFFFF" opacity=".4" rx="3"/><rect width="40" height="6" x="762" y="559" fill="#FFFFFF" opacity=".4" rx="3"/><rect width="40" height="6" x="872" y="553" fill="#FFFFFF" opacity=".4" rx="3"/><rect width="40" height="6" x="811" y="547" fill="#FFFFFF" opacity=".4" rx="3"/></g></svg>
```
Comment (weerd) [inline public/svg/500.svg:1]: The `public/svg` directory and  SVG files were added in like 5.8 but then removed by 6.0.

File: public/svg/503.svg [hunk 10]
```diff
@@ -0,0 +1 @@
....69 129.55 0 263 0c250 0 331.46 162.6 530 175 107.42 6.71 163-26.77 231-58.92V319H0z"/><path fill="url(#i)" d="M353.02 319H0V68.93C67.12 35.69 129.55 0 263 0c71.14 0 151.5 12.76 151.5 70.5 0 54.5-45.5 79.72-112.5 109-82.26 35.95-54.57 111.68 51.02 139.5z"/><path fill="url(#j)" d="M353.02 319H0v-14.8l302-124.7c-82.26 35.95-54.57 111.68 51.02 139.5z"/></g><circle cx="414" cy="799" r="70" fill="#FFFFFF" opacity=".1"/><circle cx="479" cy="745" r="30" fill="#FFFFFF" opacity=".1"/><g fill="#FFFFFF" opacity=".15" transform="translate(49 214)"><path d="M554.67 131.48a9.46 9.46 0 0 1 13.33 0 9.46 9.46 0 0 0 13.33 0l13.33-13.24a28.39 28.39 0 0 1 40 0l10 9.93a14.2 14.2 0 0 0 20 0 14.2 14.2 0 0 1 20 0l.6.6a31.8 31.8 0 0 1 9.4 22.56H548v-3.84c0-6.01 2.4-11.78 6.67-16.01zM751 8.25c11.07-11 28.93-11 40 0l10 9.94a14.19 14.19 0 0 0 20 0 14.19 14.19 0 0 1 20 0 16.36 16.36 0 0 0 21.3 1.5l8.7-6.47a33.47 33.47 0 0 1 40 0l4.06 3.03A39.6 39.6 0 0 1 931 48H731c0-12.72 8.93-28.75 20-39.75zM14.1 75.14l.9-.9a21.29 21.29 0 0 1 30 0 21.29 21.29 0 0 0 30 0l10-9.93a35.48 35.48 0 0 1 50 0l15 14.9a14.2 14.2 0 0 0 20 0 14.2 14.2 0 0 1 20 0c6.4 6.35 10 15 10 24.02V109H0c0-12.71 5.07-24.9 14.1-33.86z"/></g></g></svg>
```
Comment (weerd) [inline public/svg/503.svg:1]: The `public/svg` directory and  SVG files were added in like 5.8 but then removed by 6.0.

File: config/contentful.php [hunk 11]
```diff
@@ -1,32 +1,37 @@
...
 
-    /*
-     * Sets the locale in which to fetch content by default. NULL means the space'd default locale will be used
-     */
-    'delivery.defaultLocale' => 'en-US',
+        /*
+        * Sets the locale in which to fetch content by default. NULL means the space'd default locale will be used
+        */
+        'defaultLocale' => 'en-US',
+    ],
```
Comment (weerd) [inline config/contentful.php:36]: So glad this was fixed!

File: .phpunit.result.cache (hunk 3 above)
Comment (DFurnes) [reply .phpunit.result.cache]: It should! Removed & ignored in 7efbbfbc.

File: app/Repositories/QueriesContentful.php (hunk 4 above)
Comment (DFurnes) [reply app/Repositories/QueriesContentful.php]: Fixed in ce014c02! :a:

File: composer.json (hunk 6 above)
Comment (DFurnes) [reply composer.json:29]: It did look really cool! I'm fine with adding it, although a little curious what the reason was for removing now...

File: public/svg/403.svg (hunk 7 above)
Comment (DFurnes) [reply public/svg/403.svg]: Whiplash! Removed in 454ee2ef.

File: public/svg/404.svg (hunk 8 above)
Comment (DFurnes) [reply public/svg/404.svg]: Removed in 454ee2ef.

File: public/svg/500.svg (hunk 9 above)
Comment (DFurnes) [reply public/svg/500.svg]: Removed in 454ee2ef.

File: public/svg/503.svg (hunk 10 above)
Comment (DFurnes) [reply public/svg/503.svg]: Removed in 454ee2ef.

File: config/contentful.php (hunk 11 above)
Comment (DFurnes) [reply config/contentful.php:36]: Agreed!

File: composer.json (hunk 6 above)
Comment (weerd) [reply composer.json:29]: All I found was [this](https://github.com/laravel/laravel/commit/f053116c5680e77c3a6c73afd193984a17ea482d) with the commit title saying it didn't work in 5.9, but I tried it on 6.0 and works gangbusters 🤷‍♂️

**Pull request merged**